org/mozilla/jss/netscape/security/x509/AlgorithmId.java

```diff
@@ -177,12 +189,88 @@ public static AlgorithmId parse(byte[] val)
     }
 
     /**
-     * Constructs a parameterless algorithm ID.
+     * Constructs an algorithm ID with a fully encoded params object
+     *
+     * @ param oid the identifier for the algorithm
+     * @ param params the fully encoded AlgorithmIdentifier Object
+     */
+
```

Comment:
Same as below; NSAE and IO Exception should be in the constructor and remove space.